examples/diagnostic_test.go

```diff
@@ -0,0 +1,27 @@
+package examples
+
+import (
+	"bytes"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCredentialsErrorNotDuplicated(t *testing.T) {
+	var outputBuf bytes.Buffer
+	test := getBaseOptions().
+		With(integration.ProgramTestOptions{
+			Dir:           filepath.Join(getCwd(t), "bucket-yaml"),
+			Quick:         true,
+			Stderr:        &outputBuf,
+			ExpectFailure: true,
+			ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
```

Comment:
I think this is not working in CI because getBaseOptions() actually configures valid creds. You need to simulate no creds situation.

Reply:
Thanks for the suggestion. I think CI and the tests in general depend on the ambient aws cli being logged in and don't configure explicit credentials. I opted for explicit invalid credentials in the test.

Reply:
Ah I see, nevermind! My suggestion is off then. Thank you!